src/formpack/reporting/export.py

```diff
@@ -480,104 +518,221 @@ def format_line(line, sep, quote):
+                                accuracy,
+                            ],
+                        },
+                        "properties": {
+                            "question_name": "response value",
+                        },
+                    },
+                    {
+                        ...
+                    },
+                    ...
+                ],
             }
         """
 
         # Consider the first section only (discard repeating groups)
         first_section_name = get_first_occurrence(self.sections.keys())
         labels = self.labels[first_section_name]
+        sections = self.sections[first_section_name]
 
-        # Manually write the beginning and end of the GeoJSON file so that we
-        # can yield one GeoJSON `Feature` object at a time
+        # Set up some convenient properties when `yield`ing
         feature_array_preamble = '\n'.join([
             '{',
             '"type": "FeatureCollection",',
             '"name": "{name}",'.format(name=first_section_name),
             '"features": [',
         ])
         feature_array_epilogue = '\n]\n}'
-        yield feature_array_preamble
+        array_preamble = '[\n'
+        array_epilogue = '\n]'
+        comma_new_line = ',\n'
+        new_line = '\n'
+
+        if flatten:
+            yield feature_array_preamble
+        else:
+            yield array_preamble
 
-        self.reset() # since we're not using `parse_submissions()`
+        self.reset()  # since we're not using `parse_submissions()`
 
         first = True
         for submission in submissions:
+            if not flatten:
+                if first:
+                    yield feature_array_preamble
+                    first = False
+                else:
+                    yield comma_new_line + feature_array_preamble
+
             # We need direct access to the field objects (available inside the
             # version) and the unformatted submission data
             version = self.get_version_for_submission(submission)
             formatted_chunks = self.parse_one_submission(submission, version)
             if not formatted_chunks:
                 continue
 
-            # Find the requested field
             all_fields = version.sections[first_section_name].fields.values()
-            geo_fields = [f for f in all_fields if f.name == geo_question_name]
-            try:
-                geo_field = geo_fields[0]
-            except IndexError:
-                # Requested field doesn't exist in the form version used by
-                # this submission; skip it
-                continue
+            all_geo_fields = [
+                f for f in all_fields if f.data_type in GEO_QUESTION_TYPES
+            ]
+            all_geo_field_names = [f.name for f in all_geo_fields]
+
+            # Iterate through all geo questions and format only those that have
+            # been answered
+            first_geo = True
+            for geo_field in all_geo_fields:
```

Comment:
It seems like you could have one `for field in all_fields` loop instead of three loops (two list comprehensions and one for loop). I can't imagine an appreciable difference in performance, though, so if you find the way you have it now to be more readable than a single loop, just leave it as is

Reply:
That loop is getting pretty convoluted as it is, so I think I'd rather keep them as comprehensions to make it easier to read 👍